spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/ActuatorOperations.java

```diff
@@ -29,19 +33,31 @@
 	 * @param guid unique id for the app instance.
 	 * @param endpoint the endpoint path relative to the base actuator URL for the instance, with or without preceding '/'.
 	 * @param responseType the expected response type.
+	 * @param requestHeaders optional request headers.
 	 * @return the contents as the given type.
 	 */
-	<T> T getFromActuator(String deploymentId, String guid, String endpoint, Class<T> responseType);
+	<T> T getFromActuator(String deploymentId, String guid, String endpoint, Class<T> responseType,
+			Optional<HttpHeaders> requestHeaders);
+
+	default <T> T getFromActuator (String deploymentId, String guid, String endpoint, Class<T> responseType) {
```

Comment:
I would be in favor of only providing a single method that requires the optional headers. This will not be a primary API where convenience like this would matter. One less API to maintain. 

Whatever we land on, same goes for POST variant.

Reply:
Let's just leave for now since its a default. Lets copy/pasta the javadocs in the merge.